packages/mui-joy/src/FormLabel/FormLabel.tsx

```diff
@@ -72,7 +81,14 @@ const FormLabel = React.forwardRef(function FormLabel(inProps, ref) {
   };
 
   const classes = useUtilityClasses();
-  const externalForwardedProps = { ...other, component, slots, slotProps };
+  const externalForwardedProps = {
+    ...other,
+    component,
+    slots,
+    slotProps,
+    ...(htmlFor !== undefined && { htmlFor }),
+    ...(id !== undefined && { id }),
+  };
```

Comment:
`externalForwardedProps` is utilized in the asterisk slot, and applying `htmlFor` and `id` should only be done on the root slot. This appears to be incorrect.

The suggested correction is as follows:
```diff
--- a/packages/mui-joy/src/FormLabel/FormLabel.tsx
+++ b/packages/mui-joy/src/FormLabel/FormLabel.tsx
@@ -62,7 +62,15 @@ const FormLabel = React.forwardRef(function FormLabel(inProps, ref) {
     name: 'JoyFormLabel',
   });

-  const { children, component = 'label', slots = {}, slotProps = {}, ...other } = props;
+  const {
+    children,
+    component = 'label',
+    htmlFor,
+    id,
+    slots = {},
+    slotProps = {},
+    ...other
+  } = props;
   const formControl = React.useContext(FormControlContext);
   const required = inProps.required ?? formControl?.required ?? false;

@@ -76,8 +84,8 @@ const FormLabel = React.forwardRef(function FormLabel(inProps, ref) {

   const [SlotRoot, rootProps] = useSlot('root', {
     additionalProps: {
-      htmlFor: formControl?.htmlFor,
-      id: formControl?.labelId,
+      htmlFor: htmlFor ?? formControl?.htmlFor,
+      id: id ?? formControl?.labelId,
     },
     ref,
     className: classes.root,
```

Reply:
I wasn't sure what was the difference between `externalForwardedProps` and `additionalProps`. Fixed